rclcpp_lifecycle/src/lifecycle_node.cpp

```diff
@@ -555,6 +555,28 @@ LifecycleNode::deactivate(LifecycleNodeInterface::CallbackReturn & cb_return_cod
     lifecycle_msgs::msg::Transition::TRANSITION_DEACTIVATE, cb_return_code);
 }
 
+const State &
+LifecycleNode::raise_error()
+{
+  if(get_current_state().id() == lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE)
```

Comment:
I do not think this needs to be checked here. it will check the current state and get the reference to `rcl_lifecycle_transition_t` in rcl state_machine.